Dotnet8 metrics in otel format (#1)
lmolkova/semantic-conventions: docs/dotnet/dotnet-aspnetcore-metrics.md

# Semantic Conventions for ASP.NET Core metrics

**Status**: [Experimental, Feature-freeze][DocumentStatus]

This document defines semantic conventions for ASP.NET Core metrics.

> <!-- toc -->

Reviewer: The build system automatically creates a TOC from H2's. Building one manually is a maintenance problem. Titles frequently change.

<!-- semconv metric.aspnetcore.routing.match_attempts(metric_table) -->
| Name     | Instrument Type | Unit (UCUM) | Description    |
| -------- | --------------- | ----------- | -------------- |
| `aspnetcore.routing.match_attempts` | Counter | `{match_attempt}` | Number of requests that were attempted to be matched to an endpoint. [1] |

**[1]:** Meter name: `Microsoft.AspNetCore.Routing`; Added in: ASP.NET Core 8.0
<!-- endsemconv -->

<!-- semconv metric.aspnetcore.routing.match_attempts(full) -->
| Attribute  | Type | Description  | Examples  | Requirement Level |
|---|---|---|---|---|
| `aspnetcore.routing.match_status` | string | Match result - success or failure | `success`; `failure` | Required |
| `aspnetcore.routing.is_fallback_route` | boolean | A value that indicates whether the matched route is a fallback route. | `True` | Conditionally Required: if and only if a route was successfully matched. |
| `http.route` | string | The matched route (path template in the format used by the respective server framework). See note below [1] | `/users/:userID?`; `{controller}/{action}/{id?}` | Conditionally Required: if and only if a route was successfully matched. |

> **[1]:** MUST NOT be populated when this is not supported by the HTTP server framework as the route attribute should have low-cardinality and the URI path can NOT substitute it.

Reviewer: ```suggestion
**[1]:** MUST NOT be populated when this isn't supported by the HTTP server framework. The route attribute should have low-cardinality and the URI path can ***not*** substitute it.
```
